src/test/java/org/broadinstitute/hellbender/utils/read/ReadUtilsUnitTest.java

```diff
@@ -387,4 +390,47 @@ public void testGetReferenceIndex( final GATKRead read, final SAMFileHeader head
         Assert.assertEquals(ReadUtils.getReferenceIndex(read, header), expectedReferenceIndex, "Wrong reference index for read");
         Assert.assertEquals(ReadUtils.getMateReferenceIndex(read, header), expectedMateReferenceIndex, "Wrong reference index for read's mate");
     }
+
+    @DataProvider(name="createSAMWriter")
+    public Object[][] createSAMWriterData() {
+        return new Object[][] {
+                // Note: count_bases==query sorted, unindexed==coord sorted. We expect to silently fail
+                // to create an index if createIndex is true and sort order is not coord.
+                {new File(getToolTestDataDir(), "count_bases.bam"),   false,  true, true, false},
+                {new File(getToolTestDataDir(), "unindexed.bam"),     false,  true, true, true},
+                {new File(getToolTestDataDir(), "count_bases.bam"),   true,   true, true, false},
+                {new File(getToolTestDataDir(), "unindexed.bam"),     true,   true, true, true},
+                {new File(getToolTestDataDir(), "count_bases.bam"),   true,   true, false, false},
+                {new File(getToolTestDataDir(), "unindexed.bam"),     true,   true, false, true},
```

Comment:
Recommend that you copy these two files to new `query_sorted.bam` and `coordinate_sorted.bam` files, rather than re-purposing existing files from other tests, to make it clear what their purpose is.